### PR TITLE
feat(cast): add string-decode to decode string

### PR DESCRIPTION
**Pull request by jsvisa**
## Motivation

When I'm debugging some transactions that we reverted, what I got from the tx's trace as below:

```json
{
  "from": "0x7567d83b7b8d80addcb281a71d54fc7b3364ffed",
  "gas": "0x0",
  "gasUsed": "0x0",
  "to": "0x5ff0bea4603b68c7e80b8a973a666b9840db248c",
  "input": "0x6a76120200000000000000000000000034053113b5ddc6fb368e4d028c1bbbcb85c14ef1...",
  "output": "0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000054753303235000000000000000000000000000000000000000000000000000000",
  "error": "execution reverted",
  "calls": [
    {
      "from": "0x5ff0bea4603b68c7e80b8a973a666b9840db248c",
      "gas": "0xbbf0",
      "gasUsed": "0x24cc",
      "to": "0x632cd990566bc2cc888f75c9bbe8b9185a91d24d",
      "input": "0x6a76120200000000000000000000000034053113b5ddc6fb368e4d028c1bbbcb85c14ef1...",
      "output": "0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000054753303235000000000000000000000000000000000000000000000000000000",
      "error": "execution reverted",
      "type": "DELEGATECALL"
    }
  ],
  "value": "0x0",
  "type": "CALL"
}
```

And this output is ABI-encode of `Error(string)`, so I think we can add a simple command to decode the ABI-string into human readable msg, for this case:

```bash
$ ./target/release/cast 
 string-decode 0x88c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000054753303235000000000000000000000000000000000000000000000000000000

"GS025"
```

Inspired from https://ethereum.stackexchange.com/a/77530

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
sounds reasonable to me
wdyt @grandizzy

**Review by grandizzy — CHANGES_REQUESTED** [review]
makes sense, thank you! @jsvisa could you please also add a test, here's a sample: https://github.com/foundry-rs/foundry/blob/9d74675bae8bfbd83428ff1343cbe2ae206c3383/crates/cast/tests/cli/main.rs#L1403-L1405

File: crates/cast/bin/args.rs [hunk 1]
```diff
@@ -536,6 +536,19 @@ pub enum CastSubcommand {
...
 
+    /// Decode string
+    ///
+    /// Similar to `calldata-decode --input`, but the function argument is a `string`
+    #[command(visible_aliases = &["--string-decode", "sd"])]
+    StringDecode {
+        /// The ABI-encoded string
+        data: String,
+
+        /// Print the decoded calldata as JSON.
```
Comment (zerosnacks) [inline crates/cast/bin/args.rs:547]:
```suggestion
        /// Print the decoded string as JSON.
```

File: crates/cast/bin/args.rs [hunk 2]
```diff
@@ -536,6 +536,19 @@ pub enum CastSubcommand {
         json: bool,
     },
 
+    /// Decode string
+    ///
+    /// Similar to `calldata-decode --input`, but the function argument is a `string`
+    #[command(visible_aliases = &["--string-decode", "sd"])]
+    StringDecode {
+        /// The ABI-encoded string
+        data: String,
```
Comment (zerosnacks) [inline crates/cast/bin/args.rs:545]:
```suggestion
        /// The ABI-encoded string.
        data: String,
```

File: crates/cast/bin/args.rs [hunk 3]
```diff
@@ -536,6 +536,19 @@ pub enum CastSubcommand {
         json: bool,
     },
 
+    /// Decode string
```
Comment (zerosnacks) [inline crates/cast/bin/args.rs:539]:
```suggestion
    /// Decode ABI-encoded string.
```

**Review by zerosnacks — APPROVED** [review]
Applied minor documentation fixes, proposed goal is reasonable + implementation LGTM

**Review by grandizzy — APPROVED** [review]

**Pull request merged**